pkgs/development/libraries/libavif/default.nix

```
@@ -35,13 +39,21 @@ stdenv.mkDerivation rec {
   ];
 
   buildInputs = [
+    gdk-pixbuf
     libaom
     zlib
     libpng
     libjpeg
     dav1d
   ];
 
+  postInstall = ''
+    mkdir -p ${placeholder "gdkPixbufLoader"}/share/thumbnailers
+    cat ${./thumbnailer} | sed "s|@bindir@|${gdk-pixbuf}/bin|g" > ${placeholder "gdkPixbufLoader"}/share/thumbnailers/libavif.thumbnailer
```

You should be able to use

```suggestion
    substitute "${./thumbnailer}" "${placeholder "gdkPixbufLoader"}/share/thumbnailers/libavif.thumbnailer" --subst-var-by bindir "${gdk-pixbuf}/bin"
```